### ISSUE TITLE
Build failure related to treeForFastboot in v0.16

### ISSUE THREAD

**Issue by simonihmig** [embroider-build/embroider#448]
The recent release causes `ember-bootstrap`'s embroider scenario to fail with a `Funnel: Expected Broccoli node, got null for inputNodes[0]` error: https://github.com/kaliber5/ember-bootstrap/pull/1075/checks?check_run_id=670579781#step:5:138

I confirmed this locally see the following log output, and that this does not happen for v0.15.
<details>
  <summary>Error log</summary>

```
=================================================================================

ENV Summary:

  TIME: Wed May 13 2020 13:45:03 GMT+0200 (Central European Summer Time)
  TITLE: ember
  ARGV:
  - /Users/simonihmig/.volta/tools/image/node/12.16.2/6.14.4/bin/node
  - /Users/simonihmig/Projects/ember-bootstrap/node_modules/.bin/ember
  - serve
  EXEC_PATH: /Users/simonihmig/.volta/tools/image/node/12.16.2/6.14.4/bin/node
  TMPDIR: /var/folders/q3/yzggf_l965zbzps_bk8xr37c0000gn/T
  SHELL: /bin/zsh
  PATH:
  - /var/folders/q3/yzggf_l965zbzps_bk8xr37c0000gn/T/yarn--1589370294679-0.4868756004200283
  - /Users/simonihmig/Projects/ember-bootstrap/node_modules/.bin
  - /Users/simonihmig/.config/yarn/link/node_modules/.bin
  - /Users/simonihmig/.volta/tools/image/node/12.16.2/6.14.4/libexec/lib/node_modules/npm/bin/node-gyp-bin
  - /Users/simonihmig/.volta/tools/image/node/12.16.2/6.14.4/lib/node_modules/npm/bin/node-gyp-bin
  - /Users/simonihmig/.volta/tools/image/node/12.16.2/6.14.4/bin/node_modules/npm/bin/node-gyp-bin
  - /Users/simonihmig/.volta/tools/image/node/12.16.2/6.14.4/bin
  - /Users/simonihmig/.volta/tools/image/yarn/1.21.1/bin
  - /Users/simonihmig/.yarn/bin
  - /Users/simonihmig/.cargo/bin
  - /usr/local/bin
  - /usr/bin
  - /bin
  - /usr/sbin
  - /sbin
  - /Users/simonihmig/Projects/ember-bootstrap/node_modules/.bin
  PLATFORM: darwin x64
  FREEMEM: 353779712
  TOTALMEM: 17179869184
  UPTIME: 1694082
  LOADAVG: 3.68310546875,2.81689453125,3.04052734375
  CPUS:
  - Intel(R) Core(TM) i7-6700HQ CPU @ 2.60GHz - 2600
  - Intel(R) Core(TM) i7-6700HQ CPU @ 2.60GHz - 2600
  - Intel(R) Core(TM) i7-6700HQ CPU @ 2.60GHz - 2600
  - Intel(R) Core(TM) i7-6700HQ CPU @ 2.60GHz - 2600
  - Intel(R) Core(TM) i7-6700HQ CPU @ 2.60GHz - 2600
  - Intel(R) Core(TM) i7-6700HQ CPU @ 2.60GHz - 2600
  - Intel(R) Core(TM) i7-6700HQ CPU @ 2.60GHz - 2600
  - Intel(R) Core(TM) i7-6700HQ CPU @ 2.60GHz - 2600
  ENDIANNESS: LE
  VERSIONS:
  - ares: 1.15.0
  - brotli: 1.0.7
  - cldr: 36.0
  - http_parser: 2.9.3
  - icu: 65.1
  - llhttp: 2.0.4
  - modules: 72
  - napi: 5
  - nghttp2: 1.40.0
  - node: 12.16.2
  - openssl: 1.1.1e
  - tz: 2019c
  - unicode: 12.1
  - uv: 1.34.2
  - v8: 7.8.279.23-node.34
  - zlib: 1.2.11

ERROR Summary:

  - broccoliBuilderErrorStack: [undefined]
  - code: [undefined]
  - codeFrame: [undefined]
  - errorMessage: Funnel: Expected Broccoli node, got null for inputNodes[0]
  - errorType: [undefined]
  - location:
    - column: [undefined]
    - file: [undefined]
    - line: [undefined]
  - message: Funnel: Expected Broccoli node, got null for inputNodes[0]
  - name: TypeError
  - nodeAnnotation: [undefined]
  - nodeName: [undefined]
  - originalErrorMessage: [undefined]
  - stack: TypeError: Funnel: Expected Broccoli node, got null for inputNodes[0]
    at Funnel.Plugin (/Users/simonihmig/Projects/ember-bootstrap/node_modules/broccoli-plugin/index.js:23:13)
    at new Funnel (/Users/simonihmig/Projects/ember-bootstrap/node_modules/@embroider/compat/node_modules/broccoli-funnel/index.js:57:10)
    at V1Addon.buildTreeForFastBoot (/Users/simonihmig/Projects/ember-bootstrap/node_modules/@embroider/compat/src/v1-addon.js:613:20)
    at V1Addon.build (/Users/simonihmig/Projects/ember-bootstrap/node_modules/@embroider/compat/src/v1-addon.js:702:14)
    at V1Addon.<anonymous> (/Users/simonihmig/Projects/ember-bootstrap/node_modules/typescript-memoize/dist/memoize-decorator.js:67:52)
    at V1Addon.get v2Trees (/Users/simonihmig/Projects/ember-bootstrap/node_modules/@embroider/compat/src/v1-addon.js:375:30)
    at V1Addon.<anonymous> (/Users/simonihmig/Projects/ember-bootstrap/node_modules/typescript-memoize/dist/memoize-decorator.js:67:52)
    at /Users/simonihmig/Projects/ember-bootstrap/node_modules/@embroider/compat/src/v1-addon.js:366:99
    at V1Addon.throughTreeCache (/Users/simonihmig/Projects/ember-bootstrap/node_modules/@embroider/compat/src/v1-addon.js:392:20)
    at V1Addon.get v2Tree [as v2Tree] (/Users/simonihmig/Projects/ember-bootstrap/node_modules/@embroider/compat/src/v1-addon.js:366:21)

=================================================================================

```
</details>

The error stack reveals that the new `treeForFastBoot` method is involved, so apparently related to #444. I did *not* thoroughly investigate this, but it *seems* that:
* `undefined` (or some other falsy value) is passed here to `Funnel`: https://github.com/embroider-build/embroider/pull/444/files#diff-1c814e9cbc187d03ea0129c9a5b6f1a3R728
* so `this.customizes('treeForFastBoot')` is not correctly guarding against this, as the addon has no special FastBoot customizations (no `addon/fastboot` and no `treeForFastBoot` hook)

**Comment by ef4** [comment]
Thanks @simonihmig! I am very glad to have addons catching bugs like this in CI.

> so this.customizes('treeForFastBoot') is not correctly guarding against this, as the addon has no special FastBoot customizations (no addon/fastboot and no treeForFastBoot hook)

This code runs for all your dependencies too. It looks like ember-cli-content-security-policy does have a custom fastboot tree and the bug is happening there.

**Comment by jelhan** [comment]
> It looks like ember-cli-content-security-policy does have a custom fastboot tree and the bug is happening there.

Will try to add a Embroider scenario in ember-cli-content-security-policy to confirm. Not that familiar with Embroider and Ember CLI build hooks to be honest. But will see if I can fix it.

**Comment by simonihmig** [comment]
@jelhan AFAIU it's not the fault of `ember-cli-content-security-policy`. And apparently @ef4 already fixed this in #451! :tada:

I guess we can close this now!?

**Comment by ef4** [comment]
Yeah, it's ok to close this, though there is also a second open bug I found in your example (#450).
